mlflow/utils/uri.py

```diff
@@ -281,3 +282,35 @@ def dbfs_hdfs_uri_to_fuse_path(dbfs_uri):
         )
 
     return _DBFS_FUSE_PREFIX + dbfs_uri[len(_DBFS_HDFS_URI_PREFIX) :]
+
+
+def resolve_uri_if_local(local_uri):
+    """
+    if `local_uri` is passed in as a relative local path, this function
+    resolves it to absolute path relative to current working directory.
+
+    :param local_uri: Relative or absolute path or local file uri
+
+    :return: a fully-formed absolute uri path
+    """
+    from mlflow.utils.file_utils import local_file_uri_to_path
+
+    if local_uri is not None and is_local_uri(local_uri):
+        scheme = get_uri_scheme(local_uri)
+        cwd = pathlib.Path.cwd()
+        local_path = local_file_uri_to_path(local_uri)
+        if not pathlib.Path(local_path).is_absolute():
+            if scheme == "":
+                return str(cwd.joinpath(local_path))
+            local_uri_split = urllib.parse.urlsplit(local_uri)
+            resolved_absolute_uri = urllib.parse.urlunsplit(
+                (
+                    local_uri_split.scheme,
+                    None,
```

Comment:
Hmmm...`is_local_uri` appears to be broken for the case where `local_uri` is a `file://` URI containing a hostname that isn't localhost, e.g. `file://<some IP address>/path/to/file`. As a result, passing `None` for the host could cause problems here.

Can we update `is_local_uri` to check that the hostname component of the URI is empty before returning True?

Reply:
Can we add a test case for `is_local_uri` with a URI like `file://<some IP address>/path/to/file` to make sure it returns `False`?